lib/aws_ssm_provider.ex

```diff
@@ -20,8 +20,12 @@ defmodule AwsSsmProvider do
          {:ok, configs} <- parse_ssm_config(ssm_configs, app_name) do
       Config.Reader.merge(initial_configs, configs)
     else
-      {:error, err} -> throw({:error, err})
-      err -> throw({:unown_error, err})
+      {:error, err} ->
+        msg = error_message(err, config_path)
+        throw({:error, msg})
```

Comment:
Throwing is the right thing to do here, since that's how `Config.Provider.boot` expects to be informed of errors (see https://github.com/elixir-lang/elixir/blob/99f504e9dcb47875f18dc0b9d52d05d2da47485f/lib/elixir/lib/config/provider.ex#L206-L212)

Otherwise, we end up with elixir choking on our `:error` atom:

```
{"ERROR! Expected provider AwsSsmProvider to return a list, got: :error
```